safari/build-safari.py

```diff
@@ -0,0 +1,148 @@
+import json
+import os
+import plistlib
+from sys import platform
+
+
+def update_extension_plist(operating_system: str):
+    """Set the correct URLs in extension plist
+
+    Apple requires plist level updates to extensions. This took a while to figure out.
+    :return: None
+    """
+    with open(f"{operating_system}/recap/recap Extension/Info.plist", "rb") as f:
+        p = plistlib.loads(f.read())
+    p["NSExtension"]["SFSafariPageProperties"] = {
+        "Level": "Some",
+        "Allowed Domains": [
+            "ecf.*",
+            "ecf-train.*",
+            "pacer.*",
+            "*.uscourts.gov",
+        ],
+    }
+    with open(f"{operating_system}/recap/recap Extension/Info.plist", "wb") as f:
+        plistlib.dump(p, fp=f)
+
+
+def update_manifest_files(manifest, operating_system: str) -> None:
+    """Update the manifest files for iOS and macOS.
+
+    :param manifest: The Manifest File as JSON
+    :param operating_system: The OS to update the manifest for
+    :return: None
+    """
+
+    os.system(
+        f"cd {operating_system}/recap && xcrun agvtool new-version {manifest['version']}"
+    )
+    os.system(
+        f"cd {operating_system}/recap && xcrun agvtool new-marketing-version {manifest['version']}"
+    )
+
+    manifest["permissions"] = [
+        "*://*.uscourts.gov/",
+        "notifications",
+        "storage",
+        "unlimitedStorage",
+        "activeTab",
+        "cookies",
+    ]
+    if operating_system == "iOS":
+        manifest["background"]["persistent"] = False
+
+    with open(
+        f"{operating_system}/recap/recap Extension/Resources/manifest.json", "w"
+    ) as f:
+        json.dump(manifest, f, indent=2)
+
+
+def update_content_delegate(operating_system: str) -> None:
+    """Update the respective content delegate files for safari.
+
+    :return:None
+    """
+    content_delegate = (
+        f"{operating_system}/recap/recap Extension/Resources/content_delegate.js"
+    )
+
+    with open(content_delegate, "r") as f:
+        content = f.read()
+    content = content.replace(
+        "(navigator.userAgent.indexOf('Chrome') < 0)",
+        "((navigator.userAgent.indexOf('Safari') < 0 + navigator.userAgent.indexOf('Chrome')) < 0)",
```

Comment:
is there a way to update content_delegate.js in `main` instead of tweaking it in this fragile way?

Reply:
I wanted to avoid modifying anything in main.  leave the gospel as gospel

Reply:
Seems surgical and safe enough to go for it so you can eliminate the fragility and code here.